base/event/FairEventHeader.cxx

```diff
@@ -16,7 +16,8 @@
 
 void FairEventHeader::Register(Bool_t Persistence)
 {
-    FairRootManager::Instance()->Register("EventHeader.", "EvtHeader", this, Persistence);
+    FairRootManager::Instance()->Register(
+        "EventHeader.", "EvtHeader", this, (FairRootManager::Instance()->GetSink() ? Persistence : false));
```

Comment:
I am not so happy about FairRootManager::Instance being used more. We possibly should look at the callers and implement the check there.

That said, I am okay with this for now.

Reply:
So would you prefer to apply the check `GetSink` in the places where we actually `Register` the branches, i.e. in `FairRunAna*` and `FairMCApplication`?

Reply:
Yes, I would prefer that. It would make things more straight IMHO.

And for FairMCApplication (that's FairRunSim, right?) I would then ask: Is running a simulation without a sink a supported use case? If not, then not checking the sink and letting RegisterImpl throw a warning would actually be fine (as we would warn the user about a not really supported use case).